### PR TITLE
EPMRPP-38629. React. Fixed overall statistics widget redirect to

### PR DESCRIPTION
**Pull request by twinfreaks**

### CODE REVIEW

File: app/src/components/widgets/charts/launchExecutionAndIssueStatistics/issueStatisticsChart.jsx [hunk 1]
```diff
@@ -131,25 +136,26 @@ export class IssueStatisticsChart extends Component {
...
     const nameConfig = getItemNameConfig(d.id);
     const id = getResult(widget).id;
-    const defaultParams = this.getDefaultLinkParams(id);
+    const appliedWidgetFilterId = widget.appliedFilters[0].id;
+    const activeFilter =
+      launchFilters.filter((filter) => filter.id === appliedWidgetFilterId)[0] || {};
+    const defaultParams = this.getDefaultLinkParams(activeFilter.id);
     const defectLocators = getDefectTypeLocators(nameConfig, defectTypes);
-
     const link = getDefectLink({ defects: defectLocators, itemId: id });
```
[the remark this answers is not in the document]
Comment (DiscoElevator) [reply app/src/components/widgets/charts/launchExecutionAndIssueStatistics/issueStatisticsChart.jsx]: Also, should redirection in `Launch execution and issue statistic` widget work the same way?

[the remark this answers is not in the document]
Comment (twinfreaks) [reply app/src/components/widgets/charts/launchExecutionAndIssueStatistics/issueStatisticsChart.jsx]: Requirements were clarified with Yauheniya Labanava and now it should work this way. If the filter was not added to the page then we need to be redirected to the All launches page without any filter.

File: app/src/components/widgets/charts/launchExecutionAndIssueStatistics/launchExecutionChart.jsx [hunk 2]
```diff
@@ -128,20 +133,39 @@ export class LaunchExecutionChart extends Component {
...
+      const activeFilter = launchFilters.filter((filter) => filter.id === appliedWidgetFilterId)[0];
+      const activeFilterId = activeFilter && activeFilter.id;
+      defaultParams = this.getDefaultParamsOverallStatisticsWidget(activeFilterId);
+    } else {
+      defaultParams = this.getDefaultParamsLaunchExecutionWidget(id);
+    }
 
     const link = getStatisticsLink(nameConfig.defectType.toUpperCase());
+
     this.props.navigate(Object.assign(link, defaultParams));
```
Comment (DiscoElevator) [inline app/src/components/widgets/charts/launchExecutionAndIssueStatistics/launchExecutionChart.jsx]: You don't need statistics link when navigating to launches

File: app/src/components/widgets/charts/launchExecutionAndIssueStatistics/launchExecutionChart.jsx [hunk 3]
```diff
@@ -128,20 +133,38 @@ export class LaunchExecutionChart extends Component {
...
+
+  getDefaultParamsLaunchExecutionWidget = (id) => ({
     payload: {
       projectId: this.props.project,
-      filterId: 'all',
-      testItemIds,
+      filterId: ALL,
+      testItemIds: id,
     },
     type: TEST_ITEM_PAGE,
```
Comment (AmsterGet) [inline app/src/components/widgets/charts/launchExecutionAndIssueStatistics/launchExecutionChart.jsx]: When this widget used in the `overallStatisticsDonut` it shouldn't redirect to the step level (see the task requirements https://jiraeu.epam.com/browse/EPMRPP-38629).
This widget used in the `launchExecutionAndIssueStatistice` and as a part of `overallStatistics`.
Please provide the proper mechanism to manage this difference

Comment (AmsterGet) [reply app/src/components/widgets/charts/launchExecutionAndIssueStatistics/launchExecutionChart.jsx]: My inattention. Sorry for misleading

File: app/src/components/widgets/charts/launchExecutionAndIssueStatistics/launchExecutionChart.jsx [hunk 4]
```diff
@@ -46,6 +48,7 @@ const getResult = (widget) => widget.content.result[0] || widget.content.result;
   (state) => ({
     project: activeProjectSelector(state),
     getStatisticsLink: (name) => statisticsLinkSelector(state, { statuses: [name] }),
+    launchFilters: launchFiltersSelector(state),
```
[the remark this answers is not in the document]
Comment (AmsterGet) [reply app/src/components/widgets/charts/launchExecutionAndIssueStatistics/launchExecutionChart.jsx]: Not critical, so leave as now.

**Pull request merged**